econml/drlearner.py

```diff
@@ -928,6 +928,7 @@ def fit(self, Y, T, X=None, W=None, *, sample_weight=None, sample_var=None, infe
         if sample_weight is not None and inference is not None:
             warn("This estimator does not yet support sample variances and inference does not take "
                  "sample variances into account. This feature will be supported in a future release.")
+        Y, T, X, W, sample_weight, sample_var = check_input_arrays(Y, T, X, W, sample_weight, sample_var)

```

Comment:
Likewise, why only the SparseLinear version?  If it's because it's needed for the check on the next line, then maybe only run check_input_arrays on X and T and add a comment explaining.

Reply:
Same as above.